### PR TITLE
Improve recovery from ZRS abends in VSCode

### LINKED ISSUE #548: Unrecoverable Error Requires VS Code Reload
> Related, these seems like an unrecoverable error.  You cannot open another folder and ZNP hangs Zowe Explorer window with a busy icon indefinitely.  You must reload the VS Code window to "fix" this problem.   

 _Originally posted by @dkelosky in [#543](https://github.com/zowe/zowe-native-proto/issues/543#issuecomment-3271460593)_

### PR DESCRIPTION
**Pull request by MarkAckert**
**What It Does**
Fixes #548 and will touch on #871 via improved error messaging.

If you want to test this PR: revert commit `9409ee3ce0aac4fa62bb5e88d7b9e456074cec7b` and `afb0fcb7c1c8ce8fdf4f20a2282b3c73d9025612`. This adds a `tool abend` command to `zowex`, and changes the USS copy functionality in Zowe Explorer so it invokes the abend command rather than an actual copy. Build everything, debug the zowe explorer, and whenever you copy and paste a file in the USS File tree, you will trigger an abend and recovery, or you can run `Zowe-SSH: abend` from the command palette.

This PR captures unrecoverable errors thrown by the zowex server component and replaces the error text with a human-friendly summary and recommended actions.

The error handler from `SshClientCache` is used extensively and the error handler is invoked a little more in `ZSshClient` compared to before. If we want to remove usage of the error handler than Zowe Explorer's error correlator needs the ability to pass control back to the correlator via a hook.

To-do list:
- ~Revert `uss copy` action to actually copy and not abend intentionally~
- ~Revert `Zowe-SSH: abend` from command palette~
- ~Revert `tool abend` command~
- ~Confirm error handling for timeouts in `ZSshClient`~
- ~Confirm profile re-use in Reload ZRS~ 
- ~Handle edge-case: client is already disposed when error triggers (is this possible?)~ (No / we swallow such an edge case)
- ~Write test cases~
   
**How to Test**
See `What It Does` section.

**Review Checklist**
I certify that I have:
- [x] tested my changes
- [x] added/updated automated tests
- [x] updated the changelog
- [x] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

**Additional Comments**
n/a

### CODE REVIEW

File: packages/vsce/src/SshClientCache.ts [hunk 1]
```diff
@@ -135,8 +210,92 @@ export class SshClientCache extends vscode.Disposable {
...
+        if (clientSession.status === ServerStatus.RESTARTING) {
+            return;
+        }
+
+        const isFatal = SshClientCache.ERROR_SNIPPETS.FATAL.some((item) => errorMsg.includes(item));
+        const isTimeout = SshClientCache.ERROR_SNIPPETS.TIMEOUT.some((item) => errorMsg.includes(item));
+        const isUnsupported = SshClientCache.ERROR_SNIPPETS.UNSUPPORTED.some((item) => errorMsg.includes(item));
+
+        if (isFatal) {
+            if (clientSession) {
```
Comment (traeok) [inline packages/vsce/src/SshClientCache.ts]: Since we have a `!` assertion on line 220, the condition for `if (clientSession)` can be removed as we've already asserted that the session is not falsy.
